onnxruntime/core/providers/xnnpack/detail/utils.cc

```diff
@@ -267,9 +268,17 @@ std::unique_ptr<IndexedSubGraph::MetaDef> FuseActivation(const NodeUnit& node_un
             ORT_ENFORCE(utils::HasExternalData(value) == false,
                         "External data is not supported for the scalar min/max Clip values");
 
-            value_to_set = utils::HasRawData(value)
-                               ? *reinterpret_cast<const float*>(value.raw_data().data())
-                               : value.float_data()[0];
+            int32_t arg_type;
+            if (GetType(arg, arg_type) && arg_type == ONNX_NAMESPACE::TensorProto_DataType_FLOAT16) {
```

Comment:
What if `GetType(arg, arg_type)` failed here?

Reply:
Generally type info is always available, so I think this is ok. Shape info may be missing depending on the model. 

The Conv op looks to be setup to allow fp32, u8, s8 and optionally fp16. Should this also handle u8 and s8 or should ClipReluChecker limit fusion to fp32 and fp16?

Reply:
Clip 11 only supports float number.
https://onnx.ai/onnx/operators/onnx__Clip.html#id10
Next, we should support higher versions Clip. :)

Reply:
FYI.
So far, CPU EP Clip only supports float too.
https://github.com/microsoft/onnxruntime/blob/c6ba7edd830087bc52311a3b10b1f0692ef64b3b/onnxruntime/core/optimizer/utils.cc#L335-L349

Reply:
cc @snnn

Reply:
I'd leave the core Clip fusion as-is for now. Can be a separate PR if we think there's a use-case that would benefit.

Are you planning on updating ClipReluChecker to limit the types?

Reply:
I may need more time to understand ClipQuantFusion
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/optimizer/qdq_transformer/clip_quantizelinear.cc
But for the known reason, I have no idea about the next plan.

Reply:
I think ClipQuantFusion is a separate topic as that's about ignoring a Clip or Relu when the Q zp and scale make it redundant.

I was asking if the XNNPACK EP ClipReluChecker needs to be updated to either limit the types it allows, or whether FuseActivation needs to handle u8 or s8 input for the Clip min/max.

This has no checks on types:

https://github.com/microsoft/onnxruntime/blob/2d05c4bcd940aa25561ed7de26481f219618dd7a/onnxruntime/core/providers/xnnpack/detail/node_support_checker.cc#L42-L44

But FuseActivation always uses a float in the activation params and with this PR is explicitly only checking for fp32 and fp16.

e.g. if there's a Conv node with u8 or s8 input it looks like ClipReluChecker will allow the activation, but FuseActivation won't do the right thing as the Clip min/max would be u8 or s8.